content_scripts/CameraVis.js

```diff
@@ -90,10 +89,13 @@ void main() {
   `}
   float z = depth - 1.0;
 
+  float XtoZ = 1920.0 / focalLength.x; // 1448.24976; // width over focal length
+  float YtoZ = 1080.0 / focalLength.y; // 1448.24976;
+
   // Projection code by @kcmic
   pos = vec4(
-    (position.x / width - 0.5) * z * XtoZ,
-    (position.y / height - 0.5) * z * YtoZ,
+    (position.x - principalPoint.x) / width * z * XtoZ,
```

Comment:
This is correct but for clarity I would suggest to move conversion of focal length to new resolution to the preparation stage (same as for principal point).
So it will simplify to: (position.x - principalPoint.x) / focalLength.x * z